lib/OpenQA/Schema/Result/JobTemplates.pm

```diff
@@ -65,6 +65,10 @@ __PACKAGE__->has_many(
     settings        => 'OpenQA::Schema::Result::JobTemplateSettings',
     job_template_id => {order_by => {-asc => 'key'}});
 
+sub name {
+    $self->get_column('name') || $self->test_suite->name;
```

Comment:
I think having this function makes sense but I'd prefer if it had an explicit `return`. And have you tested that overriding works? If not you maybe need to define the override as `around 'name' => sub {`.